nums/numpy/api/creation.py

```diff
@@ -0,0 +1,1548 @@
+      diagonal, whose values are equal to one.
+
+    See Also
+    --------
+    identity : (almost) equivalent function
+    diag : diagonal 2-D array from a 1-D array specified by the user.
+
+    Notes
+    -----
+    Only k==0 is currently supported.
+
+    Examples
+    --------
+    The doctests shown below are copied from NumPy.
+    They won’t show the correct result until you operate ``get()``.
+
+    >>> nps.eye(2, dtype=int).get()  # doctest: +SKIP
+    array([[1, 0],
+           [0, 1]])
+    >>> nps.eye(3, k=1).get()  # doctest: +SKIP
+    array([[0.,  1.,  0.],
+           [0.,  0.,  1.],
+           [0.,  0.,  0.]])
+    """
+    app = _instance()
+    if k != 0:
+        raise NotImplementedError("Only k==0 is currently supported.")
+    if M is None:
+        M = N
+    shape = (N, M)
+    block_shape = app.get_block_shape(shape, dtype)
+    return app.eye(shape, block_shape, dtype)
+
+
+def diag(v: BlockArray, k=0) -> BlockArray:
+    """Extract a diagonal or construct a diagonal array.
+
+    This docstring was copied from numpy.diag.
+
+    Some inconsistencies with the NumS version may exist.
+
+    See the more detailed documentation for ``numpy.diagonal`` if you use this
+    function to extract a diagonal and wish to write to the resulting array;
+    whether it returns a copy or a view depends on what version of numpy you
+    are using.
+
+    Parameters
+    ----------
+    v : BlockArray
+        If `v` is a 2-D array, return a copy of its `k`-th diagonal.
+        If `v` is a 1-D array, return a 2-D array with `v` on the `k`-th
+        diagonal.
+    k : int, optional
+        Diagonal in question. The default is 0. Use `k>0` for diagonals
+        above the main diagonal, and `k<0` for diagonals below the main
+        diagonal.
+
+    Returns
+    -------
+    out : BlockArray
+        The extracted diagonal or constructed diagonal array.
+
+    See Also
+    --------
+    diagonal : Return specified diagonals.
+    trace : Sum along diagonals.
+    triu : Upper triangle of an array.
+
+    Notes
+    -----
+    offset != 0 is currently not supported.
+
+    out is currently not supported.
+
+    axis1 != 0 or axis2 != 1 is currently not supported.
+
+    Examples
+    --------
+    The doctests shown below are copied from NumPy.
+    They won’t show the correct result until you operate ``get()``.
+
+    >>> x = nps.arange(9).reshape((3,3))  # doctest: +SKIP
+    >>> x.get()  # doctest: +SKIP
+    array([[0, 1, 2],
+           [3, 4, 5],
+           [6, 7, 8]])
+
+    >>> nps.diag(x).get()  # doctest: +SKIP
+    array([0, 4, 8])
+
+    >>> nps.diag(nps.diag(x)).get()  # doctest: +SKIP
+    array([[0, 0, 0],
+           [0, 4, 0],
+           [0, 0, 8]])
+    """
+    app = _instance()
+    if k != 0:
+        raise NotImplementedError("Only k==0 is currently supported.")
+    return app.diag(v)
+
+
+def trace(a: BlockArray, offset=0, axis1=0, axis2=1, dtype=None, out=None):
```

Comment:
This is linear algebra (to be put under la.py or whatever we decide to call that module).